tools/cuds-prioritized-attribution/terraform/main.tf

```diff
@@ -0,0 +1,185 @@
+resource "google_project_iam_binding" "custom_role_binding" {
+  members = ["serviceAccount:${google_service_account.cud_service_account.email}"]
+  role    = "projects/${var.project_id}/roles/${google_project_iam_custom_role.cud_iam_role.role_id}"
+}
+
+# Grant Cloud Composer Role to service account
+resource "google_project_iam_binding" "composer_binding" {
+  members = ["serviceAccount:${google_service_account.cud_service_account.email}"]
+  role    = "roles/composer.worker"
+}
+
+# Grant BigQuery Permissions for Service Account on Corrected Dataset
+resource "google_bigquery_dataset" "corrected_dataset" {
+  dataset_id = "${var.corrected_dataset_id}"
+  location   = "${var.billing_export_location}"
+
+  access {
+    role          = "WRITER"
+    user_by_email = "${google_service_account.cud_service_account.email}"
+  }
+}
+
+# Create bucket for Composer temporary file store
+resource "google_storage_bucket" "commitment_file_store" {
+  name     = "${var.project_id}-cud-correction-commitment-data"
+  location = "${var.billing_export_location}"
+}
+
+# Create Composer Environment
+resource "google_composer_environment" "env" {
+  provider = "google-beta"
+  name     = "cud-correction-env"
+  region   = "${var.region}"
+  depends_on = ["google_project_service.composerapi",
+    "google_project_iam_binding.custom_role_binding",
+    "google_project_iam_binding.composer_binding",
+    "google_storage_bucket.commitment_file_store"
+  ]
+
+  config {
+    node_config {
+      zone            = "${var.zone}"
+      service_account = "${google_service_account.cud_service_account.email}"
+    }
+    software_config {
+      python_version = 3
+      airflow_config_overrides = {
+        update-pypi = "requirements.txt"
+      }
+
+      env_variables = {
+        project_id                  = "${var.project_id}"
+        billing_export_table_name   = "${var.billing_export_table_path}"
+        corrected_dataset_id        = "${var.corrected_dataset_id}"
+        corrected_table_name        = "${var.corrected_table_name}"
+        commitments_table_name      = "${var.commitment_table_path}"
+        enable_cud_cost_attribution = "${var.enable_cud_cost_attribution}"
+        cud_cost_attribution_option = "${var.cud_cost_attribution_option}"
+      }
+    }
+  }
+}
+
+
+# Upload dependencies folder to DAG folder in Composer Bucket
+resource "google_storage_bucket_object" "dag_init_upload" {
+  bucket = "${element(split("/dags", element(split("gs://", google_composer_environment.env.config.0.dag_gcs_prefix), 1)), 0)}"
+  name   = "dags/dependencies/__init__.py"
+  source = "../composer/dependencies/__init__.py"
+}
+
+# Upload dependencies folder to DAG folder in Composer Bucket
+resource "google_storage_bucket_object" "dag_billingoutput_upload" {
+  bucket = "${element(split("/dags", element(split("gs://", google_composer_environment.env.config.0.dag_gcs_prefix), 1)), 0)}"
+  name   = "dags/dependencies/billingoutput.py"
```

Comment:
note you should add a `.airflowignore` file to ignore your dependencies directory so the airflow scheduler doesn't waste cycles trying to parse non-dag-definition files.

Reply:
Hmm.. the .airflowignore would not upload them at all? Or just not parse them? Airflow needs to know about the dependencies directory to get the relevant SQL queries, used by the DAG.

Reply:
Sorry I should have included some context links in my original comment [`.airflowignore`](https://airflow.apache.org/docs/stable/concepts.html#airflowignore) just tells the scheduler what files under the `dags/` directory tree it can skip in trying to parse as a DAG definition file.

Note, that the scheduler / webserver spins up a separate process for every file  in this tree that it "thinks might be a DAG" (relatively expensive) so if we can save cycles here it improves the performance of the scheduler loop ([deeper dive wiki on scheduler](https://cwiki.apache.org/confluence/display/AIRFLOW/Scheduler+Basics) and relevant [source](https://github.com/apache/airflow/blob/e1465182ac0b058078aaf47d93225ccd98dce414/airflow/utils/file.py#L93) that walks the tree respecting airflowignore).

Reply:
done